### PR TITLE
add imp/exp command for dashboard

### PR DESCRIPTION
**Pull request by kengo0924**
**What this PR does / why we need it**:
update dashboard command to using new dashboard API that are widget style

* changes
  * add import command
  * update export command

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #[PUT ISSUE NUMBER HERE]

**Special notes for your reviewer** *(optional)*:

### CODE REVIEW

File: cmd/dashboard/root.go [hunk 1]
```diff
@@ -1,31 +1,73 @@
...
+
 // dashboardCmd represents the dashboardCmd command
 var dashboardCmd = &cobra.Command{
 	Use:   "dashboard",
 	Short: "Perform operations related to dashboard of Datadog",
 }
 
+// getJSONDataFromAPI **WORKAROUND** get JSON from dashboard api
+func getJSONDataFromAPI(path string) []byte {
+	baseurl := "https://api.datadoghq.com/api/v1/"
```
Comment (toVersus) [inline cmd/dashboard/root.go]: I think `baseurl` should be defined as a global constant.

File: cmd/dashboard/root.go [hunk 2]
```diff
@@ -1,31 +1,73 @@
...
+	param := "?api_key=" + viper.GetString("api_key") + "&application_key=" + viper.GetString("app_key")
```
Comment (toVersus) [inline cmd/dashboard/root.go]: Don't compose a query parameter in hand. Use [url](https://golang.org/pkg/net/url/) package instead.

See: https://stackoverflow.com/a/26987017

File: cmd/dashboard/root.go [hunk 3]
```diff
@@ -1,31 +1,73 @@
...
+}
+
 // dashboardCmd represents the dashboardCmd command
 var dashboardCmd = &cobra.Command{
 	Use:   "dashboard",
 	Short: "Perform operations related to dashboard of Datadog",
 }
 
+// getJSONDataFromAPI **WORKAROUND** get JSON from dashboard api
+func getJSONDataFromAPI(path string) []byte {
```
Comment (toVersus) [inline cmd/dashboard/root.go]: I think we should move this method to `datadog` package. Also, rename the method to `fetchRawData` or something.

File: cmd/dashboard/root.go [hunk 4]
```diff
@@ -1,31 +1,73 @@
...
+
+	var boards MyDashboards
+	err := json.Unmarshal([]byte(jd), &boards)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return boards.MyDashboards
+
+	// err := cli.doJsonRequest("GET", "/v1/dashboard/", nil, &boards)
```
Comment (toVersus) [inline cmd/dashboard/root.go]: Remove this line!

File: cmd/dashboard/root.go [hunk 5]
```diff
@@ -1,31 +1,73 @@
...
+	}
+
+	return jd
+}
+
+// GetAllDashboards **WORKAROUND** get all dashboard with widget form
+func GetAllDashboards() []datadog.Board {
+	jd := getJSONDataFromAPI("dashboard")
+
+	var boards MyDashboards
```
Comment (yukirii) [inline cmd/dashboard/root.go]: You can Unmarshall without temporary struct type (MyDashboards) by using `map[string][]datadog.Board`.

```go
// example
var boards map[string][]datadog.Board
err := json.Unmarshal([]byte(jd), &boards)
return boards["dashboards"]
```

File: cmd/dashboard/root.go (hunk 1 above)
Comment (kengo0924) [reply cmd/dashboard/root.go]: thanks.
I will delete this func if it implemented by go-datadog-api module( like GetDashboards), so I think no need to modify.

File: cmd/dashboard/root.go (hunk 3 above)
Comment (kengo0924) [reply cmd/dashboard/root.go]: https://github.com/tani-yu/dogleash/pull/43#discussion_r308535140

same.

**Comment by kengo0924** [comment]
thank you very much for your comments. 👍

File: cmd/dashboard/import.go [hunk 6]
```diff
@@ -24,9 +23,51 @@ var dashboardImportCmd = &cobra.Command{
...
+			}
+		}
 	},
 }
 
 func init() {
 	dashboardCmd.AddCommand(dashboardImportCmd)
 }
+
+// Check if there is the same id and name
```
Comment (tani-yu) [inline cmd/dashboard/import.go]: 【nits】I think this comment is not correct. Because this function don't check name.

**Pull request merged**